exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusUnitsHelper.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.prometheus;
+
+import io.opentelemetry.api.internal.StringUtils;
+import java.util.regex.Pattern;
+
+/**
+ * A utility class that is used to maintain mappings between OTLP unit and Prometheus units. The
+ * list of mappings is adopted from <a
+ * href="https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/prometheus/normalize_name.go#L30">OpenTelemetry
+ * Collector Contrib</a>.
+ *
+ * @see <a
+ *     href="https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#units-and-base-units">OpenMetrics
+ *     specification for units</a>
+ * @see <a href="https://prometheus.io/docs/practices/naming/#base-units">Prometheus best practices
+ *     for units</a>
+ */
+public final class PrometheusUnitsHelper {
+
+  private static final Pattern INVALID_CHARACTERS_PATTERN = Pattern.compile("[^a-zA-Z0-9]");
+  private static final String CHARACTERS_BETWEEN_BRACES_REGEX =
+      "\\{(.*?)}"; // matches all characters between {}
+
+  private PrometheusUnitsHelper() {
+    // Prevent object creation for utility classes
+  }
+
+  /**
+   * A utility function that returns the equivalent Prometheus name for the provided OTLP metric
+   * unit.
+   *
+   * @param rawMetricUnitName The raw metric unit for which Prometheus metric unit needs to be
+   *     computed.
+   * @param metricType The {@link PrometheusType} of the metric whose unit is to be converted.
+   * @return the computed Prometheus metric unit equivalent of the OTLP metric unit.
+   */
+  public static String getEquivalentPrometheusUnit(
+      String rawMetricUnitName, PrometheusType metricType) {
+    if (StringUtils.isNullOrEmpty(rawMetricUnitName)) {
+      return rawMetricUnitName;
+    }
+
+    // special case
+    if (rawMetricUnitName.equals("1") && metricType == PrometheusType.GAUGE) {
+      return "ratio";
+    }
+
+    String convertedMetricUnitName = rawMetricUnitName;
+    // Drop units specified between curly braces
+    convertedMetricUnitName = removeUnitPortionInBrackets(convertedMetricUnitName);
+
+    // Handling for the "per" unit(s), e.g. foo/bar -> foo_per_bar
+    convertedMetricUnitName = convertRateExpressedToPrometheusUnit(convertedMetricUnitName);
+
+    // Converting abbreviated unit names to full names
+    return cleanUpString(getPrometheusUnit(convertedMetricUnitName));
+  }
+
+  private static String convertRateExpressedToPrometheusUnit(String rateExpressedUnit) {
```

Comment:
While not strictly required, can you add javadoc to this method?

It helps a lot for others following the algorithm.